eclair-core/src/main/scala/fr/acinq/eclair/blockchain/bitcoind/rpc/BasicBitcoinJsonRPCClient.scala

```diff
@@ -18,18 +18,25 @@ package fr.acinq.eclair.blockchain.bitcoind.rpc
 
 import com.softwaremill.sttp._
 import com.softwaremill.sttp.json4s._
+import fr.acinq.bitcoin.ByteVector32
 import fr.acinq.eclair.KamonExt
 import fr.acinq.eclair.blockchain.Monitoring.Metrics
-import org.json4s.DefaultFormats
-import org.json4s.JsonAST.JValue
+import org.json4s.{CustomSerializer, DefaultFormats}
+import org.json4s.JsonAST.{JString, JValue}
 import org.json4s.jackson.Serialization
 
 import scala.concurrent.{ExecutionContext, Future}
 
 class BasicBitcoinJsonRPCClient(user: String, password: String, host: String = "127.0.0.1", port: Int = 8332, ssl: Boolean = false)(implicit http: SttpBackend[Future, Nothing]) extends BitcoinJsonRPCClient {
 
+  // necessary to properly serialize ByteVector32 into String readable by bitcoind
+  object ByteVector32Serializer extends CustomSerializer[ByteVector32](_ => ( {
```

Comment:
Do we have a test that verifies we correctly serialize in this client?

Reply:
Yes, [this test](https://github.com/ACINQ/eclair/blob/5eaee701e65fb6407b849dee5cf9b0b084193ce6/eclair-core/src/test/scala/fr/acinq/eclair/blockchain/bitcoind/BitcoinCoreWalletSpec.scala#L169) uses the [rollback](https://github.com/ACINQ/eclair/pull/1248/files#diff-f0e4b1153995aaf8ef920c12f6c4c36bR73) which in turn uses `Utxo` with a `ByteVector32` in it, the test fails if we don't correctly serialize the bytevector.